src/address.c

```diff
@@ -163,26 +153,8 @@ int generateDerivedHDPubkey(const char* wif_privkey_master, char* p2pkh_pubkey)
     memcpy(prefix, wif_privkey_master, 1);
     const dogecoin_chainparams* chain = memcmp(prefix, "d", 1) == 0 ? &dogecoin_chainparams_main : &dogecoin_chainparams_test;
 
-    size_t strsize = 128;
-    char str[strsize];
-
-    /* if nothing is passed in use internal variables */
-    if (p2pkh_pubkey) {
-        memcpy(str, p2pkh_pubkey, strsize);
-    }
-
-    dogecoin_hdnode node;
-    dogecoin_hdnode_deserialize(wif_privkey_master, chain, &node);
-
-    dogecoin_hdnode_get_p2pkh_address(&node, chain, str, strsize);
-
-    /* pass back to external variable if exists */
-    if (p2pkh_pubkey) {
-        memcpy(p2pkh_pubkey, str, strsize);
-    }
-
-    /* reset internal variables */
-    memset(str, 0, strlen(str));
+    /* generate p2pkh from private key */
+    generatePrivPubKeypair(wif_privkey_master, p2pkh_pubkey, memcmp(prefix, "d", 1) == 0 ? false : true);
```

Comment:
I believe this line is causing incorrect address output, sorry my test cases did not catch that. The generated private keys start with Q like a regular priv/pub keypair rather than prefixed by dgpv as HD keys should. The verifyHDMasterPubKeypair function relies on dogecoin_hdnode_deserialize() to verify the pair, and when this function is given a regular private key it does not error out but rather evaluates to the same public address each time ("D8xB9XuiWygAuMbYBziYRx6ZEddm3Mbas1")  which is what caused python unit tests to fail, as well as more in-depth C unit tests

Reply:
good catch. thanks for pr!